Modules/main.c

```diff
@@ -1804,8 +1790,15 @@ config_init_utf8_mode(_PyCoreConfig *config)
 
 
 static _PyInitError
-config_read_env_vars(_PyCoreConfig *config)
+config_read_env_vars(_PyCoreConfig *config, int ignore_cmdline)
```

Comment:
I added the ignore_cmdline to be 100% sure that I don't change the behaviour of Py_Main with this patch, and haven't checked yet if this is really necessary.

Reply:
Provided we are handling the `-I` option correctly and are also not looking at `argv` unless using `Py_Main`, it shouldn't be necessary.

Reply:
I've talked with Victor about this issue (without looking at this code in detail), the master branch contains a much better implementation of this included much improved unit tests.  

I won't update this pull request until he has had time to think about this issue.